source/Octopus.Tentacle.Client/Scripts/ScriptExecutionOrchestrator.cs

```diff
@@ -73,21 +76,34 @@ public async Task<ScriptStatusResponseV2> ExecuteScript(CancellationToken script
             {
                 try
                 {
-                    scriptStatusResponse = await rpcCallExecutor.ExecuteWithRetries(
-                        nameof(scriptServiceV2.StartScript),
-                        ct =>
-                        {
-                            ++startScriptCallCount;
-                            return scriptServiceV2.StartScript(startScriptCommand, new HalibutProxyRequestOptions(ct));
-                        },
-                        logger,
-                        // If we are cancelling script execution we can abandon a call to start script
-                        // If we manage to cancel the start script call we can walk away
-                        // If we do abandon the start script call we have to assume the script is running so need
-                        // to call CancelScript and CompleteScript
-                        abandonActionOnCancellation: true,
-                        clientOperationMetricsBuilder,
-                        scriptExecutionCancellationToken).ConfigureAwait(false);
+                    ScriptStatusResponseV2 StartScriptAction(CancellationToken ct)
+                    {
+                        ++startScriptCallCount;
+                        return scriptServiceV2.StartScript(startScriptCommand, new HalibutProxyRequestOptions(ct));
+                    }
+
+                    if (settings.AllowRetries)
+                    {
+                        scriptStatusResponse = await rpcCallExecutor.ExecuteWithRetries(
+                            nameof(scriptServiceV2.StartScript),
+                            StartScriptAction,
+                            logger,
+                            // If we are cancelling script execution we can abandon a call to start script
+                            // If we manage to cancel the start script call we can walk away
+                            // If we do abandon the start script call we have to assume the script is running so need
+                            // to call CancelScript and CompleteScript
+                            abandonActionOnCancellation: true,
+                            clientOperationMetricsBuilder,
+                            scriptExecutionCancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        scriptStatusResponse = rpcCallExecutor.Execute(
```

Comment:
Is there a behavioural change here as were loosing `abandonActionOnCancellation: true` which isn't around retries but around how cancellation happens

Reply:
Hmm good question. I assumed that `Execute` was doing the same thing as `ExecuteWithRetries`, which is probably mostly true, but I'll check on this behaviour.